client-lite/src/util/http_agent.cpp

```diff
@@ -157,144 +184,50 @@ HRESULT HttpAgent::_CreateClient(PCSTR szUrl, PCSTR szProxyUrl) try
 {
     std::unique_lock<std::recursive_mutex> lock(_requestLock);
 
-    // We must recreate the client if either the url or the proxy url has changed
-    if ((szUrl != nullptr) && _client)
+    if (!_requestContext.curlHandle)
     {
-        const std::string currentUrl = _client->base_uri().to_string();
-        if (StringCompareCaseInsensitive(currentUrl.data(), szUrl) != 0)
-        {
-            _client.reset();
-        }
-    }
-    else if ((szProxyUrl != nullptr) && _client)
-    {
-        const std::string currentProxy = _client->client_config().proxy().address().to_string();
-        if (StringCompareCaseInsensitive(currentProxy.data(), szProxyUrl) != 0)
-        {
-            _client.reset();
-        }
-    }
-
-    if (!_client)
-    {
-        RETURN_HR_IF(E_INVALIDARG, (szUrl == nullptr));
+        RETURN_HR_IF(E_INVALIDARG, szUrl == nullptr);
 
         std::string url(szUrl);
         RETURN_HR_IF(INET_E_INVALID_URL, !ValidateUrl(url));
 
-        web::http::client::http_client_config clientConfig;
-        _SetWebProxyFromProxyUrl(clientConfig, szProxyUrl);
-
-        web::http::uri remoteUri(url);
-        _client = std::make_unique<web::http::client::http_client>(remoteUri, clientConfig);
-        DoLogVerbose("New http_client for %s", remoteUri.to_string().data());
+        _requestContext.curlHandle = curl_easy_init();
+        RETURN_HR_IF(E_OUTOFMEMORY, _requestContext.curlHandle == nullptr);
+        DoLogVerbose("New http_client for %s", szUrl);
+
+        // Set options that are generic to all requests
+        curl_easy_setopt(_requestContext.curlHandle, CURLOPT_FOLLOWLOCATION, static_cast<long>(1));
+        curl_easy_setopt(_requestContext.curlHandle, CURLOPT_MAXREDIRS, static_cast<long>(10));
```

Comment:
>10 [](http://example.com/codeflow?start=90&length=2)

-1 allows infinite redericts, 10 to avoid infinite redirecting (can such a thing happen?)